common/templates/context.go

```diff
@@ -174,6 +174,7 @@ type Context struct {
 	CurrentFrame *ContextFrame
 
 	IsExecedByLeaveMessage bool
+	IsExecedByJoinMessage  bool
```

Comment:
This should probably be a single `ExecutedFrom` field at this point. Maybe leave this to a followup cleanup PR to address though.

Reply:
I totally agree with all of that, I was actually thinking the same thing while I was working on this. I agree that it would be better addressed in a follow up PR.